src/AzureExtension/DataManager/AzureDataManager.cs

```diff
@@ -560,64 +603,328 @@ private async Task UpdateDataForPullRequestsAsync(DataStoreOperationParameters p
+                };
+
+                await UpdateDataForPullRequestsAsync(suboperationParameters);
             }
+        }
+        catch (Exception ex)
+        {
+            contextDict.Add("ErrorMessage", ex.Message);
+            SendErrorUpdateEvent(_log, this, parameters.Requestor, context, ex);
+            return;
+        }
 
-            // Convert relevant pull request items to Json.
-            dynamic pullRequestsObj = new ExpandoObject();
-            var pullRequestsObjDict = (IDictionary<string, object>)pullRequestsObj;
-            foreach (var pullRequest in pullRequests)
-            {
-                dynamic pullRequestObj = new ExpandoObject();
-                var pullRequestObjFields = (IDictionary<string, object>)pullRequestObj;
+        SendDeveloperUpdateEvent(_log, this, parameters.Requestor, context);
+    }
 
-                pullRequestObjFields.Add("Id", pullRequest.PullRequestId);
-                pullRequestObjFields.Add("Title", pullRequest.Title);
+    private async Task ProcessPullRequests(
+        VssConnection connection,
+        List<GitPullRequest>? pullRequests,
+        Project project,
+        Repository repository,
+        string loginId,
+        PullRequestView view,
+        bool isDeveloper)
+    {
+        if (pullRequests is null || pullRequests.Count == 0)
+        {
+            // If PRs were null or empty, this is a valid result, so create an empty record.
+            PullRequests.GetOrCreate(DataStore, repository.Id, project.Id, loginId, view, new JsonObject().ToJsonString());
+            return;
+        }
 
-                var creator = Identity.GetOrCreateIdentity(DataStore, pullRequest.CreatedBy, result.Connection);
-                pullRequestObjFields.Add("CreatedBy", creator);
-                pullRequestObjFields.Add("CreationDate", pullRequest.CreationDate.Ticks);
-                pullRequestObjFields.Add("TargetBranch", pullRequest.TargetRefName);
+        dynamic pullRequestsObj = new ExpandoObject();
+        var pullRequestsObjDict = (IDictionary<string, object>)pullRequestsObj;
+        var policyClient = connection.GetClient<PolicyHttpClient>();
 
-                // The Links lack an html url for these results, construct the Url.
-                var htmlUrl = $"{project.ConnectionUri}_git/{azureUri.Repository}/pullrequest/{pullRequest.PullRequestId}";
-                pullRequestObjFields.Add("HtmlUrl", htmlUrl);
+        foreach (var pullRequest in pullRequests)
+        {
+            var status = PolicyStatus.Unknown;
+            var statusReason = string.Empty;
+
+            // ArtifactId is null in the pull request object and it is not the correct object. The ArtifactId for the
+            // Policy Evaluations API is this:
+            // vstfs:///CodeReview/CodeReviewId/{projectId}/{pullRequestId}
+            var artifactId = $"vstfs:///CodeReview/CodeReviewId/{project.InternalId}/{pullRequest.PullRequestId}";

```

Comment:
Can you add the link to the documentation that contains the artifact id URL format?

Reply:
Added!